src/probnum/kernels/_exponentiated_quadratic.py

```diff
@@ -0,0 +1,68 @@
+"""Exponentiated quadratic kernel."""
+
+from typing import Optional
+
+import numpy as np
+import scipy.spatial.distance
+
+import probnum.utils as _utils
+from probnum.type import IntArgType, ScalarArgType
+
+from ._kernel import Kernel
+
+_InputType = np.ndarray
+
+
+class ExpQuad(Kernel[_InputType]):
+    """Exponentiated quadratic / RBF kernel.
+
+    Covariance function defined by :math:`k(x_0, x_1) = \\exp \\big(-\\frac{\\lVert
+    x_0 - x_1 \\rVert^2}{2l^2}\\big)`. This kernel is also known as the squared
+    exponential or radial basis function kernel.
+
+    Parameters
+    ----------
+    input_dim :
+        Input dimension of the kernel.
+    lengthscale
+        Lengthscale of the kernel. Describes the input scale on which the process
+        varies.
+
+    See Also
+    --------
+    RatQuad : Rational quadratic kernel.
+    Matern : Matern kernel.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from probnum.kernels import ExpQuad
+    >>> K = ExpQuad(input_dim=1, lengthscale=0.1)
+    >>> K(np.linspace(0, 1, 3)[:, None])
+    array([[1.00000000e+00, 3.72665317e-06, 1.92874985e-22],
+           [3.72665317e-06, 1.00000000e+00, 3.72665317e-06],
+           [1.92874985e-22, 3.72665317e-06, 1.00000000e+00]])
+    """
+
+    def __init__(self, input_dim: IntArgType, lengthscale: ScalarArgType = 1.0):
+        self.lengthscale = _utils.as_numpy_scalar(lengthscale)
+        super().__init__(input_dim=input_dim, output_dim=1)
+
+    def __call__(self, x0: _InputType, x1: Optional[_InputType] = None) -> np.ndarray:
+        # Check and reshape inputs
+        x0, x1, kernshape = self._check_and_reshape_inputs(x0, x1)
+
+        # Compute pairwise euclidean distances ||x0 - x1|| / l
+        if x1 is None:
+            pdists = scipy.spatial.distance.squareform(
+                scipy.spatial.distance.pdist(x0 / self.lengthscale, metric="euclidean")
+            )
+        else:
+            pdists = scipy.spatial.distance.cdist(
+                x0 / self.lengthscale, x1 / self.lengthscale, metric="euclidean"
+            )
+
+        # Compute kernel matrix
+        kernmat = np.exp(-(pdists ** 2) / 2.0)
+
+        return Kernel._reshape_kernelmatrix(kernmat, newshape=kernshape)
```

Comment:
what does this do? Reshape the output if it is a scalar? Because the remaining shape should be handled correctly anyway, no? Please correct me if I am wrong.

also, why do you call this method as a staticmethod here? `self._reshape...` should work as well (or make it a function to start with?) That could be personal preferance on my side though, so it's fair if you disagree.

Reply:
It reshapes the output to the correct shape as determined by the input and output dimension of the kernel. This allows you to implement a kernel with an efficient way to compute the kernel matrix without having to think about what the right output shape is. All you need to know if you implement `__call__` is that you get one or two arrays N x d and you need to return some array. The correct shape is then determined by `_reshape_kernelmatrix`. The function is static because it doesn't rely on `self`. It's just `reshape` with a corner case for scalars.

Reply:
Thanks a lot for the thorough answer! 

My comment about the staticmethod was more like "why is this a static method inside `Kernel` and not a function in `_kernel.py`".  Then again, if a class has a staticmethod, I am more used to calling it with `self.some_staticmethod` instead of `MyClass.some_staticmethod` both of which work. However, there may well be good practice that I am missing though, I am not doing a lot with static methods.

Reply:
This is probably somewhat of a design choice, but this method is tailored to be used inside the `__call__` function of kernels which why semantically it makes sense to tie it to the class itself in my opinion. No other use beyond that is intended. If you think this is not valid I am happy to change it. Now calling it with `self` means that if a subclass overrides the static method, this would take precedence. To avoid bugs or at least force the developer to declare their intent I typically use the class name (since this makes explicit which method should be called). But I think this is also debatable.